.travis.yml

```diff
@@ -79,6 +80,12 @@ before_install:
   - ulimit -a
 
 install:
+  - sudo pip install flake8
+  - flake8 --version
+  - flake8 --max-line-length=120
+  - sudo pip3 install flake8
+  - python3 -m flake8 --version
+  - python3 -m flake8 --max-line-length=120
```

Comment:
I think you should wrap these commands under a condition, and only enable it on a single build (in the build matrix). Specifically, this doesn't seem to work as-is under macOS. But it's also a waste of time to run it 4 times in parallel. So, set a new environment variable, say: `flake8` and set it to true in one of the build configurations that currently is quick. Use the top one, the one with `sonar_scan=1`.